packages/flutter/test/material/search_test.dart

```diff
@@ -40,6 +40,29 @@ void main() {
     TestDefaultBinaryMessengerBinding.instance!.defaultBinaryMessenger.setMockMethodCallHandler(SystemChannels.platform, null);
   });
 
+  testWidgets('Changing query moves cursor to the end of query',
+      (WidgetTester tester) async {
+    final _TestSearchDelegate delegate = _TestSearchDelegate();
+
+    await tester.pumpWidget(TestHomePage(delegate: delegate));
+    await tester.tap(find.byTooltip('Search'));
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 300));
+
+    delegate.query = 'Foo';
+
+    final TextField textField =
+        tester.widget<TextField>(find.byType(TextField));
```

Comment:
Nit: this too should probably go on one line. Or the indent on the second line should be two spaces.